neqo-transport/benches/range_tracker.rs

```diff
@@ -0,0 +1,42 @@
+use criterion::{criterion_group, criterion_main, Criterion}; // black_box
+use neqo_transport::send_stream::RangeTracker;
+
+const CHUNK: u64 = 1000;
+const END: u64 = 100_000;
+fn build_coalesce(len: u64) -> RangeTracker {
+    let mut used = RangeTracker::default();
+    used.mark_acked(0, CHUNK as usize);
+    used.mark_sent(CHUNK, END as usize);
+    // leave a gap or it will coalesce here
+    for i in 2..=len {
+        // These do not get immediately coalesced when marking since they're not at the end or start
+        used.mark_acked(i * CHUNK, CHUNK as usize);
+    }
+    return used;
+}
+
+fn coalesce(c: &mut Criterion, count: u64) {
+    let mut used = build_coalesce(count);
+    c.bench_function(
+        &format!("coalesce_acked_from_zero {count}+1 entries"),
+        |b| {
+            b.iter(|| {
+                let used: &mut RangeTracker = &mut used;
+                used.mark_acked(CHUNK, CHUNK as usize);
+                let tail = (count + 1) * CHUNK;
+                used.mark_sent(tail, CHUNK as usize);
+                used.mark_acked(tail, CHUNK as usize);
```

Comment:
What exactly is this supposed to be doing?  
'used' is modified, and then reused in the modified state in the next iteration, etc.
After the first iteration, it keeps re-marking CHUNK to CHUNK+CHUNK-1 as acked, though they're already acked.
Unfortunately, if you restore the 'used' to the original state, or (re)clone it, that will be included in the time for the benchmark, and given cloning or restoring probably mean allocations, that may swamp the coalesce function.
I have a question in the criterion maintainers about if there's a way to have a per-iteration setup function that's not  included in the measurement

Reply:
We could build an vector of N copies of the input, and then limit iterations to N.  I'm not finding any other good options

Reply:
https://docs.rs/criterion/latest/criterion/struct.Bencher.html#method.iter_batched_ref should do it